src/Homeworks/HW6_2/Main.java

```diff
@@ -0,0 +1,148 @@
+
+                        System.out.println("Enter circle radius:");
+                        int radius = scanner.nextInt();
+                        circle = new Circle(radius);
+                        double area = circle.calculateArea();
+                        System.out.println("Circle area: " + area);
+                    } else {
+                        System.out.println("Invalid choice.");
+                    }
+                    break;
+
+                case 2:
+
+                    Triangle triangle = new Triangle(0, 0, 0);
+
+                    System.out.println("Choose what to calculate:");
+                    System.out.println("1. Perimeter");
+                    System.out.println("2. Area (using Heron's formula)");
+                    System.out.println("3. Area (using base and height)");
+                    System.out.println("4. Area (using two sides and an angle)");
+                    int triangleChoice = scanner.nextInt();
+
+                    if (triangleChoice == 1) {
+
+                        System.out.println("Enter triangle sides (a, b, c):");
+                        int sideA = scanner.nextInt();
+                        int sideB = scanner.nextInt();
+                        int sideC = scanner.nextInt();
+                        if (sideA + sideB > sideC && sideA + sideC > sideB && sideB + sideC > sideA) {
+                            triangle = new Triangle(sideA, sideB, sideC);
+                            double perimeter = triangle.calculatePerimeter();
+                            System.out.println("Triangle perimeter: " + perimeter);
+                        } else {
+                            System.out.println("Such a triangle cannot exist, enter the correct sides");
+                        }
+
+                    } else if (triangleChoice == 2) {
+
+                        System.out.println("Enter triangle sides (a, b, c):");
+                        int sideA = scanner.nextInt();
+                        int sideB = scanner.nextInt();
+                        int sideC = scanner.nextInt();
+                        if (sideA + sideB > sideC && sideA + sideC > sideB && sideB + sideC > sideA) {
+                        triangle = new Triangle(sideA, sideB, sideC);
+                        double area = triangle.calculateArea(sideA, sideB, sideC);
+                        System.out.println("Triangle area: " + area);
+                        } else {
+                            System.out.println("Such a triangle cannot exist, enter the correct sides");
+                        }
+                    } else if (triangleChoice == 3) {
+
+                        System.out.println("Enter triangle base (a):");
+                        int base = scanner.nextInt();
+                        System.out.println("Enter triangle height (h):");
+                        int height = scanner.nextInt();
+                        triangle = new Triangle(base, height);
```

Comment:
There is no necessity of using overriding constructor if you pass base and height to square method